lib/mongo/collection/view/readable.rb

```diff
@@ -193,7 +193,9 @@ def count_documents(opts = {})
           pipeline << { :'$limit' => opts[:limit] } if opts[:limit]
           pipeline << { :'$group' => { _id: 1, n: { :'$sum' => 1 } } }
 
-          opts.select! { |k, _| [:hint, :max_time_ms, :read, :collation, :session].include?(k) }
+          opts = opts.select { |k, _| [:hint, :max_time_ms, :read, :collation, :session].include?(k) }
+          opts[:collation] = @options[:collation] unless opts[:collation]
```

Comment:
I actually think this could be written in an even nicer way:
```
opts[:collation] ||= collation
```
With the `collation` method that's already defined on `view.rb`. Happy to chat more about `||=` syntax if you haven't seen it before.